core/src/input/input.stories.tsx

```diff
@@ -44,15 +64,37 @@ interface Props {
 	readOnly?: boolean;
 }
 
-export const Primary = (props: Props) => (
-	<Input
-		type={props.type}
-		style={Input.styles[props.style]}
-		size={Input.sizes[props.size]}
-		maxLength={props.maxLength}
-		disabled={props.disabled}
-		readOnly={props.readOnly}
-	/>
-);
+export const Primary = (props: Props) => {
+	const [text, setText] = useState("");
+	const onSubmit = (e: { preventDefault: () => void }) => {
+		e.preventDefault();
+		Dialog.alert(`You entered: ${text}`);
+	};
```

Comment:
Yeah this is good. But usually I'd just show the text in-place to have simpler code :))

Also the "e" type is quite complicated. For things like that I'd just have the function inlined